File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "";
    
    queue<string> cd1;
    queue<string> cd2;
    
    vector<string> ans;
    
    for(int i=0; i<cards1.size(); i++){
        cd1.push(cards1[i]);
    }
    
    for(int i=0; i<cards2.size(); i++){
        cd2.push(cards2[i]);
    }
    
    for(int i=0; i<goal.size(); i++){
        if(cd1.front() == goal[i]){
            ans.push_back(cd1.front());
            cd1.pop();
        } else if(cd2.front() == goal[i]){
            ans.push_back(cd2.front());
            cd2.pop();
        } else {
            answer = "No";
            return answer;
        }
    }
    
    int cnt = 0;
    for(int i=0; i<goal.size(); i++){
        if(ans[i] == goal[i]){
            cnt++;
        }
    }
    
    if(cnt == goal.size()) answer = "Yes";
    else answer = "No";
    
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        
        int c1 = 0, c2 = 0;
        int cnt = 0;
        while(cnt < goal.length){
            if (c1 < cards1.length && goal[cnt].equals(cards1[c1])) {
                c1++;
                cnt++;
            } else if (c2 < cards2.length && goal[cnt].equals(cards2[c2])) {
                c2++;
                cnt++;
            } else {
                break;
            }
        }
        
        if(goal.length == c1+c2) answer = "Yes";
        else answer = "No";
        
        return answer;
    }
}